### ISSUE TITLE
Add support for csl v2.x

### ISSUE THREAD

**Issue by jgreidy** [ebsco/edsapi-ruby#115]
We are unable to provide support for the MLA 9 citation format. We're trying to use the csl ~>2 gem, but that conflicts with the latest ebsco-eds gem. 

Bundler could not find compatible versions for gem "csl":
```

  In Gemfile:
    ebsco-eds was resolved to 1.1.3, which depends on
      citeproc-ruby (>= 1.0.2, ~> 1.0) was resolved to 1.0.6, which depends on
        csl (~> 1.2)

    ebsco-eds was resolved to 1.1.3, which depends on
      csl-styles (~> 2.0) was resolved to 2.0.1, which depends on
        csl (~> 2.0)


```
I made a fork and got bundler to work with this in ebsco-eds-gemspec:
```
  spec.add_dependency 'citeproc'
  spec.add_dependency 'csl', '~> 2.0'
  spec.add_dependency 'citeproc-ruby'
  spec.add_dependency 'csl-styles'
```

but the tests fail with
```
Unable to activate railties-6.1.4.1, because activesupport-6.1.5 conflicts with activesupport (= 6.1.4.1) (Gem::ConflictError)
```

I'm out of my depth.

**Comment by cclini** [comment]
Would you mind trying the code at this branch?:  https://github.com/ebsco/edsapi-ruby/tree/adding-citation-support
This adds support for Ruby 3 and csl 2.  If you have any errors, please feel free to comment here with them.  We'll be looking at putting a PR up soon.  Thanks!

**Comment by jgreidy** [comment]
I was not able to sucessfully update actionview, rails, ruby to get bundler to accept this version. I just ran out of time. Will try again later. Thanks for the update, though!

**Comment by mmabrahamson** [comment]
This should be addressed now as of #114 based on our testing.  We'll close the issue for now but if it comes back up let us know!

**Comment by jgreidy** [comment]
Hi,
I have been working on this one again. It turns out it's a big lift for us to go to activesupport (>= 7). We are currently at activesupport (6.1.7.2). I notice your master branch has activesupport '>= 5.2'. Was this activesupport change necessary on the adding-citation-support branch?

Bundler could not find compatible versions for gem "activesupport":
  In snapshot (Gemfile.lock):
    activesupport (= 6.1.7.2)

  In Gemfile:
    ebsco-eds was resolved to 1.1.4, which depends on
      activesupport (>= 7)

    rails (~> 6.1.7) was resolved to 6.1.7.2, which depends on
      activesupport (= 6.1.7.2)

---

gem 'ebsco-eds', :git => 'https://github.com/ebsco/edsapi-ruby', :branch => 'adding-citation-support'

**Comment by mmabrahamson** [comment]
This is a fair question and I'm sure it made sense to my brain a year ago.  At this point, I think we were able to incorporate most of these changes into the master branch as it's a couple versions and many commits past this now.  However, looking at master I did note that we don't have csl 2 running on it.  Before I reopen this issue and we look into addressing csl 2 again, can you verify that the master build doesn't solve the issues you were running into with citations?

**Comment by jgreidy** [comment]
Sure:
% bundle update csl-styles ebsco-eds
Fetching https://github.com/ebsco/edsapi-ruby
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies.............
Bundler could not find compatible versions for gem "csl-styles":
  In Gemfile:
    csl-styles (~> 2.0, >= 2.0.1)

    ebsco-eds was resolved to 1.1.5, which depends on
      csl-styles (~> 1.0, >= 1.0.1.5)
---
gem 'csl-styles', '~> 2.0', '>= 2.0.1'
gem 'ebsco-eds', :git => 'https://github.com/ebsco/edsapi-ruby', :branch => 'master'
